src/models/stores/serial.ts

```diff
@@ -0,0 +1,84 @@
+import { NodeChannelInfo } from "src/plugins/dataflow-tool/model/utilities/node";
+
+export class SerialDevice {
+    value: string;
+    localBuffer: string;
+    private port: SerialPort | null;
+
+    constructor() {
+      this.value = "0";
+      this.localBuffer = "";
+    }
+
+    public hasPort(){
+      return this.port !== undefined;
+    }
+
+    public async requestAndSetPort(){
+      try {
+          this.port = await navigator.serial.requestPort();
+      }
+
+      catch (error) {
+        console.error("error requesting port: ", error);
+      }
+    }
+
+    public async handleStream(channels: Array<NodeChannelInfo>){
+      await this.port?.open({ baudRate: 9600 }).catch((e: any) => console.log(e));
+
+        while (this.port?.readable) {
+
+          const textDecoder = new TextDecoderStream();
+          this.port.readable.pipeTo(textDecoder.writable);
+          const streamReader = textDecoder.readable.getReader();
+
+          try {
+              while (this.port.readable) {
+                const { value, done } = await streamReader.read();
+                if (done){
+                  break;
+                }
+                this.handleStreamObj(value, channels);
+              }
+            }
+            catch (error) {
+              console.error(error);
+            }
+            finally {
+              streamReader.releaseLock();
+            }
+          }
+    }
+
+    public handleStreamObj(value: string, channels: Array<NodeChannelInfo>){
+
+      // keep buffer from growing too large
```

Comment:
@kswenson I wonder what you think of this.  Even though we lop the discovered match off from the buffer each time, the buffer gradually grows in size and the parsing seems to get "behind" or offset over time.  I will step through with a slower example but I wonder if I am missing anything obvious?